clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -1160,12 +1158,11 @@ List<NetworkClientDelegate.UnsentRequest> drain(final long currentTimeMs) {
             failAndRemoveExpiredCommitRequests(currentTimeMs);
 
             // Add all unsent offset commit requests to the unsentRequests list
-            unsentRequests.addAll(
-                    unsentOffsetCommits.stream()
-                        .filter(request -> request.canSendRequest(currentTimeMs))
-                        .peek(request -> request.onSendAttempt(currentTimeMs))
-                        .map(OffsetCommitRequestState::toUnsentRequest)
-                        .collect(Collectors.toList()));
+            List<NetworkClientDelegate.UnsentRequest> unsentRequests = unsentOffsetCommits.stream()
+                .filter(request -> request.canSendRequest(currentTimeMs))
+                .peek(request -> request.onSendAttempt(currentTimeMs))
+                .map(OffsetCommitRequestState::toUnsentRequest)
+                .collect(Collectors.toList());
```

Comment:
We should use toCollection to define the implementation in order to avoid the possible JDK change in the future